core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -715,7 +748,8 @@ pub struct StateManagerDatabase<R> {
 
 impl ActualStateManagerDatabase {
     pub fn new(
-        root: PathBuf,
+        root_path: PathBuf,
+        checkpoints_path: PathBuf,
```

Comment:
Are there any cons of simply accepting a full `target_checkpoint_path` in each `/db-checkpoint` request?

Regarding pros that I see:
- higher flexibility for the caller
- less code impact (e.g. if someone uses `new_read_only()` constructor and does not care about snapshots, let's not bother them with the new `checkpoints_path: PathBuf` param)

Reply:
Yeah I considered that too. I went with a configurable folder for two reasons:
- IMO it's slightly easier to use
- and it's slightly more secure / less error prone (e.g. it can prevent unthoughtful calls to create a snapshot on an external drive)

both with the emphasis on "slightly".
Is it worth the price of reduced flexibility? I can be convinced that it isn't :man_shrugging:

Reply:
I see these convenience aspects, but I assume that a target user here is an above-average admin (hey - someone who DOES backups, right? :D), so they will very likely integrate via some script and they will also test their paths.
So I'd still personally pick the "less impact on the unrelated code" way.

Reply:
or we could keep a single configured directory, but have it managed by the API handler itself, and provided to state manager in each call, wdyt?

Reply:
nice, best of both worlds!

Reply:
_(done)_